File: ParityGen.java
import java.util.Scanner;


public class ParityGen {
	public static void main(String[] args) {
		//Intro
		Scanner keyboardInput = new Scanner(System.in);

		//Takes Parity Method
		System.out.println("Java");
		System.out.println("This program prints a parity bit on a binary 8-bit integer");
	    System.out.println("Even (0) or Odd (1) Parity Bit?");

		//Declare variables in the farthest scope you'll use them in
		int parityMethod;
		while (true) {
    		try {
    			parityMethod = keyboardInput.nextInt();

	    		if (parityMethod == 0) {
    				System.out.println("You have chosen the Even Parity Check");
    				break;
    			} 
	    		else if (parityMethod == 1) {
    				System.out.println("You have chosen the Odd Parity Check");
    				break;
	    		}
    			//In case input is invalid
    			else {
    				System.out.println("Please enter either 0 or 1");
				}
    		}		
	    	catch(Exception A) {
				keyboardInput.nextLine();
    			System.out.println("Please enter an integer value");
			}
    	}

		//Checks that input is legal 
		//caveman mode engaging
		String userInput;
		while (true) {
			System.out.println("Please enter 7 bits");
			//Takes User Input
			userInput = keyboardInput.next();

			//Checks Length
			boolean legalLength = false;  
			if (userInput.length() == 7){
				legalLength = true;
			}

			//checks for 1's and 0's
			//After a few months, I forgor wtf I did
			int i = 0;
			boolean legalInput = true;
			while (i < userInput.length()) {
				//char userInputChar = userInput.charAt(i);
				if ((userInput.charAt(i) != '0') && (userInput.charAt(i) != '1')) {
					legalInput = false;
				}
				i++;
			}
			
			if ((legalLength == true) && (legalInput == true)) {
				System.out.println("You have entered 7 bits");
				break;
			}
			else if ((legalLength == true) && (legalInput == false)) {
				System.out.println("You have entered 7 characters, but they're not bits");
			}
			else if ((legalLength == false) && (legalInput == true)) {
				System.out.println("You have entered an incorrect amount of bits");
			}
			else if ((legalLength == false) && (legalInput == false)) {
				System.out.println("You have entered neither 7 characters, nor bits");
			}
		}
	
		//caveman mode disengaging
		//I could've used switch statements but I tried to make it as comprehensible as possible

		//Genarating Parity Bit
		//Reusing i because I don't need to worry about it clashing with its previous instance
		int numberOfOnes = 0;
		int i = 0;
		while (i < 7) {
			if ((userInput.charAt(i) == '1')) {
				numberOfOnes++;
			}
			i++;
		}
		int parityBit = (((numberOfOnes % 2) + parityMethod) % 2);

		//Adding Parity Bit & printing
		String transmittedByte = userInput + Integer.toString(parityBit); // or String.valueOf()
		System.out.println("Transmitted byte will be " + transmittedByte);

		keyboardInput.close();


	}
}

File: ParityGen.py
print("Python")
print("This program prints a parity bit on a binary 7-bit integer")
print("Even (0) or Odd (1) Parity Method?")

parityMethod = -1
#Choosing parity method & vaildating
while True:
    try:
        parityMethod = int(input())
        if (parityMethod == 1):
            print("You have chosen an odd parity method")
            break
        elif (parityMethod == 0):
            print("You have chosen an even parity method")
            break
        else:
            print("Please enter 0 or 1")
    except:
        parityMethod = -1
        print("Please enter an integer")


userInput =  (-1)
#Choosing data & validating
while True:
    print("Please enter 7 bits")
    try:
        userInput = str(input())

        #Checking Length
        legalLength = 0
        if len(userInput) == 7:
            legalLength = 1
        
        #Checking Input
        legalInput = 0
        if ((('0' in userInput) == True) or (('1' in userInput) == True)):
            legalInput = 1
            print(legalInput)
    
        #Evaluating
        if ((legalLength == 1) and (legalInput == 1)):
            print("You have entered 7 bits")
            break
    
        elif ((legalLength == 1) and (legalInput == 0)):
            print("You have entered 7 characters, but they're not bits")
            
        elif ((legalLength == 0) and (legalInput == 1)):
            print("You have entered an incorrect amount of bits")
        
        elif ((legalLength == 0) and (legalInput == 0)):
            print("You have entered neither 7 characters, nor bits")
    
    except:
        print("Please enter 7 bits")

#Generating Parity Bit
numberOfOnes = 0
for bit in userInput:
    if (bit == 1):
        numberOfOnes += 1
parityBit = (((numberOfOnes % 2) + parityMethod) % 2)

#Printing
transmittedByte = userInput + str(parityBit)
print("Transmitted byte will be " + transmittedByte)